### ISSUE TITLE
Using Val_series for RNN or N-beats for multiple univariate time series takes forever to compute whereas when val_series isnt used it takes the usual time

### ISSUE THREAD

**Issue by Kennethfargose** [unit8co/darts#1798]
So I have been using RNN on multiple univarate data where I convert multiple columns to single univaraite and try to run different models, the models which don't need val_series work fine but when I use it the code its forever to compute I have end it. To put things in context I have used N-beats and Rnn without the val_series and they work fine N-beats gives good results and Rnn shitty. I am confused why  val_series takes so much time

**Comment by Kennethfargose** [comment]
def univariate_model_report(dataset_list, validation_start_date, model, model_name, model_kwargs):
    model = model(**model_kwargs)
    model_name = model_name

    res = []

    for dataset in dataset_list:
        ts_list = convert_multivariate_ts_to_list(dataset)

        for ts in ts_list:
           
        
            name = ts.components.item()
            ts_train, ts_test = ts.split_after(validation_start_date)
            import time
            start_time = time.time()
            model = model.fit(ts_train)
            end_time = time.time()
            time=end_time-start_time
            preds = model.predict(len(ts_test))
            res.append(pd.DataFrame(
                {
                "model_name": [model_name],
                "ts_name": [name],
                "r2": [r2_score(ts_test, preds)],
                "mae": [mae(ts_test, preds)],
                "mape": [mape(ts_test, preds)],
                "smape": [smape(ts_test, preds)],
                "marre": [marre(ts_test, preds)],
                "mase":[mase(ts_test, preds,ts_train)],
                "ope":[ope(ts_test, preds)],
                "rmsle":[rmsle(ts_test, preds)],
                "KWARGS":[model_kwargs],
                "time":[time]
                }
            ))
            

            ts.plot(label="actual")
            preds.plot(label="forecast")
            plt.legend()
    return pd.concat(res)    hey @dennisbader  this is my code  the fucntion  convert_multivariate just converts dataframe with multiple columns into unvarite time series.    and the data looks something like this  in dataframe and it is sales data



<html>
<body>


date | Store_1| Store_2| Store_3
-- | -- | -- | --
2014-12-24 | 8649.27 | 10157.870 | 8804.670
2014-12-25 | 9103.34 | 11011.585 | 10342.235
2014-12-26 | 9557.41 | 11865.300 | 11879.800
2014-12-27 | 8556.81 | 9870.420 | 9973.110


</body>
</html>

**Comment by dennisbader** [comment]
Some things you might not be aware of:
- you can fit/predict on a list of (univarate or multivariate) time series instead of doing this iteratively
- you can change how often the validation set is evaluated with `pl_trainer_kwargs` at model creation (a dict of parameters corresponding to the PyTorch Lightning args/kwargs from [here](https://lightning.ai/docs/pytorch/stable/common/trainer.html#trainer-class-api)). For example with `limit_val_batches`, `check_val_every_n_epoch `, ...
- we can only fully investigate if something is going wrong with a reproducible example (e.g. you generate some data in the code to reproduce)
